packages/multicast_dns/lib/src/packet.dart

```diff
@@ -162,7 +162,10 @@ _FQDNReadResult _readFQDN(
         final Uint8List partBytes =
             Uint8List.view(data.buffer, offset, partLength);
         offset += partLength;
-        parts.add(utf8.decode(partBytes));
+        // According to the RFC, this is supposed to be utf-8 encoded, but
+        // we should continue decoding even if it isn't to avoid dropping the
+        // rest of the data, which might still be useful.
```

Comment:
For future me: I agree with this because it should give a better error message to the end user: something like `I can't find device "Dan?s iPhone"`rather than `FormatException` or `Can't find device: null` or something.